File: Math/poj2429.java
// GCD & LCM Inverse
// This Problem is favouring the JAVA
// It runs FIVE THOUSAND MILLISECOND
// while it has been Successfully Accepted.
import java.util.Scanner;
public class Main{
    private static long gcd,lcm;
    public static void main(String[] args){
        Scanner in=new Scanner(System.in);
        while(in.hasNext()){
            gcd=in.nextLong();
            lcm=in.nextLong();
            solve();
        }
    }
    private static long GCD(long a,long b){
        return b==0?a:GCD(b,a%b);
    }
    private static void solve(){
        long prod=lcm/gcd;
        for(long i=(long)Math.floor(Math.sqrt(prod));i>0;i--){
            if(prod%i==0&&GCD(i,prod/i)==1){
                System.out.printf("%d %d\n",i*gcd,prod/i*gcd);
                return;
            }
        }
    }
}

File: Math/poj2429.cpp
// GCD & LCM Inverse

#include <cmath>
#include <ctime>
#include <iostream>

typedef long long int64;

int64 GCD(int64 a, int64 b) {
    if (b == 0) {
        return a;
    } else {
        return GCD(b, a % b);
    }
}

int main() {
    std::ios::sync_with_stdio(false);
    std::cout.tie(NULL);
    std::cin.tie(NULL);
    int64 gcd, lcm;
    while (std::cin >> gcd >> lcm) {
        for (int64 k = (int64)sqrt(lcm / gcd); k > 0; k--) {
            if ((lcm / gcd) % k == 0) {
                int64 j = lcm / gcd / k;
                if (GCD(j, k) == 1) {
                    std::cout << k * gcd << " " << j * gcd << '\n';
                    break;
                }
            }
        }
    }
    return 0;
}